classes/actions.php

```
@@ -507,6 +527,11 @@ public static function perform_moveto(array $modules, int $target): void {
                 throw new moodle_exception('sectionnotexist', 'block_massaction');
             }
 
+            // Not moving if the section is restricted.
+            if (in_array($cm->sectionnum, $sectionsrestricted)) {
+                continue;
```

Should we throw an exception of sectionrestricted here?